### PR TITLE
feat: support configurable upload rate

### LINKED ISSUE #32: support configurable UploadRate
Currently, the `UploadRate` in SessionConfig is a static value (10 * time.Seconds).

It is necessary to make this param configurable, so that the user can custom the upload rate.

### PR DESCRIPTION
**Pull request by eastfisher**
Closes #32

### CODE REVIEW

File: pyroscope/api.go [hunk 1]
```diff
@@ -59,6 +60,14 @@ func Start(cfg Config) (*Profiler, error) {
 		return nil, err
 	}
 
+	if cfg.UploadRate == 0 {
+		// For backward compatibility.
+		cfg.UploadRate = 10 * time.Second
+	} else if cfg.UploadRate <= 5*time.Second {
```
Comment (petethepig) [inline pyroscope/api.go:66]: I would drop the limit, I could see people wanting to do less than 5 seconds.

Comment (eastfisher) [reply pyroscope/api.go]: Fixed

**Comment by eastfisher** [comment]
Similar to #33

**Comment by eastfisher** [comment]
The comment above is fixed, would you please continue reviewing? ^_^

**Review by petethepig — APPROVED** [review]

**Pull request merged**